Slim/Menu/BrowseLibrary.pm

```diff
@@ -1063,26 +1064,30 @@ sub _artists {
 	my @ptSearchTags = @searchTags;
 	@ptSearchTags = grep {$_ !~ /^genre_id:/} @ptSearchTags if _getPref('noGenreFilter', $remote_library);
 
-	if ( _getPref('noRoleFilter', $remote_library) && (my (@roles) = grep /^role_id:/, @ptSearchTags) ) {
-		@ptSearchTags = grep {$_ !~ /^role_id:/} @ptSearchTags;
+	if ( _getPref('noRoleFilter', $remote_library) ) {
+		@ptSearchTags = grep {$_ !~ /^role_id:/} @ptSearchTags if $mode && $mode eq 'artists';
+	} else {
 
-		# "no role filter" means the default role list _plus_ what we specifically want
+		# if single artist list, filter is the default role list _plus_ what we specifically want
 		if ( _getPref('useUnifiedArtistsList', $remote_library) ) {
+			my @roles = grep {$_ =~ /^role_id:/} @ptSearchTags;
 			@roles = map {
 				/role_id:(.*)/;
 				Slim::Schema::Contributor->roleToType($1);
-			} @roles;
+			} @roles  if @roles;
 
-			push @roles, 'ARTIST', 'TRACKARTIST', 'ALBUMARTIST';
+			if ( $mode && $mode eq 'artists' ) {
+				push @roles, 'ARTIST', 'TRACKARTIST', 'ALBUMARTIST';
 
-			# Loop through each pref to see if the user wants to show that contributor role.
-			foreach (Slim::Schema::Contributor->contributorRoles) {
-				if (_getPref(lc($_) . 'InArtists', $remote_library)) {
-					push @roles, $_;
+				# Loop through each pref to see if the user wants to show that contributor role.
+				foreach (Slim::Schema::Contributor->contributorRoles) {
+					if (_getPref(lc($_) . 'InArtists', $remote_library)) {
+						push @roles, $_;
+					}
 				}
-			}
 
-			push @ptSearchTags, 'role_id:' . join(',', @roles);
+				push @ptSearchTags, 'role_id:' . join(',', @roles);
+			}
 		}
 	}
```

Comment:
I'm not sure I fully understand what you're doing here... your change definitely changes behaviour for the existing menu behaviour. Wouldn't it be better to use the menu mode as the first condition, and only apply the new changes if we're not dealing with one of the default artists lists? Eg. the `useUnifiedArtsitsList` should NOT be evaluated if we're dealing with a custom contributor menu mode.

Reply:
See discussion starting here: https://github.com/Logitech/slimserver/pull/953#issuecomment-1856453739

I've deliberately changed the behaviour based on that discussion.

Reply:
Ok then! Now even I remember 🤦🏻.